ios/browser/api/ads/brave_ads.mm

```diff
@@ -1422,183 +1271,116 @@ - (void)updateAdRewards {
-  const auto key = base::SysUTF8ToNSString(path);
-  const auto json = (NSString*)self.prefs[key];
-  if (!json) {
-    return absl::nullopt;
+- (void)maybeMigrateProfilePrefs {
+  NSString* legacyProfilePrefsPath =
+      [self.storagePath stringByAppendingPathComponent:@"ads_pref.plist"];
+  NSDictionary* legacyProfilePrefs = [[NSMutableDictionary alloc]
+      initWithContentsOfFile:legacyProfilePrefsPath];
+  if (!legacyProfilePrefs) {
+    return;
   }
 
-  absl::optional<base::Value> value =
-      base::JSONReader::Read(base::SysNSStringToUTF8(json));
-  if (!value || !value->is_dict()) {
-    return absl::nullopt;
-  }
+  NSLog(@"Migrating profile prefs");
 
-  return value->GetDict().Clone();
-}
+  if ([legacyProfilePrefs objectForKey:@"BATAdsEnabled"]) {
+    const BOOL isEnabled = [legacyProfilePrefs[@"BATAdsEnabled"] boolValue];
 
-- (void)setListPref:(const std::string&)path value:(base::Value::List)value {
-  std::string json;
-  if (base::JSONWriter::Write(value, &json)) {
-    const auto key = base::SysUTF8ToNSString(path);
-    self.prefs[key] = base::SysUTF8ToNSString(json);
-    [self savePref:key];
-  }
-}
+    [self setProfilePrefValue:brave_rewards::prefs::kEnabled
+                        value:base::Value(isEnabled)];
 
-- (absl::optional<base::Value::List>)getListPref:(const std::string&)path {
-  const auto key = base::SysUTF8ToNSString(path);
-  const auto json = (NSString*)self.prefs[key];
-  if (!json) {
-    return absl::nullopt;
+    [self setProfilePrefValue:brave_ads::prefs::kOptedInToNotificationAds
+                        value:base::Value(isEnabled)];
   }
 
-  absl::optional<base::Value> value =
-      base::JSONReader::Read(base::SysNSStringToUTF8(json));
-  if (!value || !value->is_list()) {
-    return absl::nullopt;
+  NSError* error = nil;
+  [[NSFileManager defaultManager] removeItemAtPath:legacyProfilePrefsPath
+                                             error:&error];
+  if (error) {
+    NSLog(@"Failed trying to remove legacy prefs: %@", error);
   }
-
-  return value->GetList().Clone();
 }
 
-- (void)clearPref:(const std::string&)path {
-  const auto key = base::SysUTF8ToNSString(path);
-  [self.prefs removeObjectForKey:key];
-  [self savePref:key];
+- (void)setProfilePrefValue:(const std::string&)path value:(base::Value)value {
+  self.profilePrefService->Set(path, value);
 }
 
-- (bool)hasPrefPath:(const std::string&)path {
-  const auto key = base::SysUTF8ToNSString(path);
-  return [self.prefs objectForKey:key] != nil;
+#pragma mark - Local state prefs
+
+- (void)initLocalStatePrefService {
+  _localStatePrefService = GetApplicationContext()->GetLocalState();
+  CHECK(_localStatePrefService);
 }
 
-- (void)setLocalStatePref:(const std::string&)path value:(base::Value)value {
-  // Not needed on iOS
+- (void)setLocalStatePrefValue:(const std::string&)path
+                         value:(base::Value)value {
+  self.localStatePrefService->Set(path, value);
 }
```

Comment:
Is there a reason we have both `setLocalStatePrefValue:value:` and `setLocalState:Value:`. The latter is the ads client bridge method I believe, but I assume we can just have 1 (same for profile prefs setter)

Reply:
Fixed